keydata.go

```diff
@@ -174,6 +148,27 @@ type KeyParams struct {
 	// device models, and also the digest algorithm used to produce the
 	// key digest.
 	SnapModelAuthHash crypto.Hash
+	PlatformName      string // Name of the platform that produced this data
+
+	// KDFAlg is the digest algorithm used to derive additional keys during
+	// the use of the created KeyData. It must match the algorithm passed to
+	// [MakeDiskUnlockKey]. The zero value here has a special meaning which
+	// is reserved to support legacy TPM2 key data files, and tells the
+	// KeyData to use the unique key as the unlock key rather than using it
+	// to derive the unlock key.
+	KDFAlg crypto.Hash
```

Comment:
I'm confused by the comment about the legacy case, as this used for creating a new KeyData and I don't see any code that special case this KDFAlg when it's null, nor I understand what goes into EncryptedPayload in that case

Reply:
So I gave it a second look and my guess would be that this is probably forgotten comment/code (maybe from an older implementation?), since we use the `version` field to differentiate the keys (which now is also marked as `omitempty`). I removed the unreachable block and the existing tests pass (as expected).

Reply:
those are files without the json wrapper?

Reply:
Thanks. I would mention tpm2.NewKeyDataFromSealedKeyObjectFile in the comment for clarity.